### PR TITLE
add create command to cli which takes json argument

### PR DESCRIPTION
**Pull request by whatsadebugger**
Not sure about my test but it passes.

### CODE REVIEW

**Comment by whatsadebugger** [comment]
I know which test broke. Im looking at it

**Comment by thodges-gh** [comment]
I receive the following error when trying to add a JobSpec:
```
 thomas@fedoratop  ~/go/src/github.com/smartcontractkit/chainlink   master  ./chainlink c '{"initiators":[{"type":"web"}],"tasks":[{"type":"HttpGet","url":"https://bitstamp.net/api/ticker/"},{"type":"JsonParse","path":["last"]},{"type":"EthBytes32"},{"type":"EthTx","address":"0x356a04bce728ba4c62a30294a55e6a8600a320b3","functionSelector":"0x609ff1bd"}]}'
json: cannot unmarshal object into Go value of type []models.JobSpec
```

I'll work with Ahmad offline to get this resolved.

File: cmd/client_test.go [hunk 1]
```diff
@@ -78,3 +78,14 @@ func TestClient_ShowJobSpec_NotFound(t *testing.T) {
...
+
+func TestClient_CreateJobSpec(t *testing.T) {
+	app, cleanup := cltest.NewApplication()
+	defer cleanup()
+
+	client, _ := cltest.NewClientAndRenderer(app.Store.Config)
+	set := flag.NewFlagSet("create", 0)
+	set.Parse([]string{"badinput"})
+	c := cli.NewContext(nil, set, nil)
+	assert.NotNil(t, client.CreateJobSpec(c))
```
Comment (dimroc) [inline cmd/client_test.go]: Good start with the specs, the flag set is always a bit clumsy.

This checks for bad input, let's get a happy path test in there as well:
```
{"initiators":["web"], "tasks":["noop"]}
```

And let's assert that a job was made:

```golang
assert.Equal(t, len(app.Store.Jobs()), 1)
```

Finally, we can wrap this all up by putting the two cases in one table driven test: https://blog.golang.org/subtests

Chainlink example: https://github.com/smartcontractkit/chainlink/blob/chores/go-report/utils/utils_test.go#L76

Let me know if you have any issues or questions 👍 
- Dimitri

**Comment by dimroc** [comment]
Great tests @whatsadebugger, thanks! 👍

**Comment by whatsadebugger** [comment]
No problem! I Was also thinking of adding a -file flag or file subcommand so you can pass in a file. @dimroc

**Comment by dimroc** [comment]
We actually have another story coming up that is in that space. @se3000
will be writing up the details as an issue in the coming days. Feel free to
jump on it.

**Comment by whatsadebugger** [comment]
Ok awesome working on it :+1:

**Pull request merged**